#167728017 Implementation of user signup feature

Wokoro opened the pull request:
**What does this PR do?**

  This is an implementation of user signup endpoint, that would allow users of the platform to create new accounts.

  **Description of Task to be completed?**

  Users should be able to register on the platform through a dedicated endpoint, upon successful registration users should receive a message of successful account creation.

  **How should this be manually tested?**

 * Clone repository
 * Install dependencies
 * Navigate to `localhost:3000/api/v1/auth/signup/` using postman
 * Provide require user signup details and send

  **Any background context you want to provide?**

  All dependency should be install.

  **What are the relevant stories?**

 [#167728015](https://www.pivotaltracker.com/story/show/167728015)

  **Screenshots (if appropriate)**
  
![signup](https://user-images.githubusercontent.com/23240530/63938754-668d2f80-ca1a-11e9-9337-5fea7e7d05e8.jpg)

  **Questions:**

none

Wokoro committed:
ft(signup):User sigup implemetation

    Implemetations for user signup endpoint.
    [Finished #167728017]

Abobos commented on .editorconfig:
```diff
@@ -0,0 +1,18 @@
+
+[{*.json, *.yml}]
+indent_style = space
+indent_size = 2
```
what's the file for?

Wokoro replied:
It is meant to work with our IDE's, so we have a unified coding style and files before pushing to git...

Wokoro replied:
On file save, the above settings are affected on the giving file...
Sorry I didn't raise that it was on my mind to....

justinefe commented on src/controllers/AuthController.js:
```diff
@@ -13,12 +13,13 @@ export default class AuthController {
-   * @return {Object} Object resoponse with current user created status
+   * @return {Object} Return sucess message and account creation status
    */
   static async signup({ body }, res, next) {
```
I think TTL requested that the static method be removed

richienabuk commented on tests/user.test.js:
```diff
@@ -11,18 +11,19 @@ chai.use(chaiHttp);
       })
       .end((err, res) => {
-        expect(res.status).eql(201);
+        expect(res.status).eql(200);
```
I think signup should return 201 not 200 because new entry is created.

Wokoro replied:
Thank's for pointing this out, changes would be effected immediately.

justinefe commented:
Cool thanks for the feedback I really don't think sign in is new entry
created like sign up.. Sign in should be OK.. 200

On Aug 31, 2019 9:28 PM, "Imo-owo Nabuk" <notifications@github.com> wrote:

> *@richienabuk* commented on this pull request.
> ------------------------------
>
> In tests/user.test.js
> <https://github.com/andela/cyclops-backend/pull/37#discussion_r319735536>:
>
> >        })
>        .end((err, res) => {
> -        expect(res.status).eql(201);
> +        expect(res.status).eql(200);
>
> I think signup should return 201 not 200 because new entry is created.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/andela/cyclops-backend/pull/37?email_source=notifications&email_token=ALXKCE6DAOQP355MBWTY7JLQHLH67A5CNFSM4ISBPFMKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCDJ5QVQ#pullrequestreview-282318934>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALXKCE7YQFUEHXQTFUYIFYTQHLH67ANCNFSM4ISBPFMA>
> .
>

Wokoro committed:
bg(code):return appropriate status code

    Changed status code for user signup from 200 to 201,
    indicating successful account creation.

chukwuemekachm reviewed (commented):
I'm curious to know what happens when a user with that email already exists?

Are we hashing a user's password before saving?

Wokoro commented:
> I'm curious to know what happens when a user with that email already exists?
> 
> Are we hashing a user's password before saving?

Currently, no checks are been done for already existing user account.
The user password is been hashed before saving to the database.

Wokoro committed:
bg(fix): Implemenation for user account check

    Implementation to check if user account exist before creating
    account.
